src/OrchardCore.Modules/OrchardCore.Redis/Caching/RedisDistributedCache.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+using OrchardCore.Modules;
+
+namespace OrchardCore.Redis.Caching
+{
+    /// <summary>
+    /// Wrapper to ensure the system stays available during Redis outages
+    /// </summary>
+    public class RedisDistributedCache : OrchardCore.Caching.Distributed.IDistributedCache
+    {
+        private readonly IDistributedCache _underlyingCache;
+        private readonly IClock _clock;
+
+        private static DateTime _retryAt;
+        private static bool _redisUnavailable = false;
+        private static object _redisUnavailableLock = new object();
+
+        public RedisDistributedCache(Microsoft.Extensions.Caching.Distributed.IDistributedCache cache, IClock clock)
+        {
+            _underlyingCache = cache;
+            _clock = clock;
+        }
+
+        public byte[] Get(string key)
+        {
+            CheckDistributedErrorRetry();
+
+            if(_redisUnavailable)
+                return Array.Empty<byte>();
```

Comment:
We have an `.editorconfig` file at the root so that e.g. under Visual studio you get some warnings about code style rules. Here, always use braces `{` and `}` for if blocks and on new lines

Reply:
The .editorconfig file isn't working locally for me which is odd.  It has the root=true and I've verified that my 'Follow project coding conventions' option is enabled.